src/Modules/Build.lua

```diff
@@ -318,8 +318,8 @@ function buildMode:Init(dbFileName, buildName, buildXML, convertBuild)
 		{ stat = "AreaOfEffectRadius", label = "AoE Radius", fmt = "d" },
 		{ stat = "BrandAttachmentRange", label = "Attachment Range", fmt = "d", flag = "brand" },
 		{ stat = "BrandTicks", label = "Activations per Brand", fmt = "d", flag = "brand" },
-		{ stat = "ManaCost", label = "Mana Cost", fmt = "d", color = colorCodes.MANA, compPercent = true, lowerIsBetter = true, condFunc = function(v,o) return o.ManaHasCost end },
-		{ stat = "LifeCost", label = "Life Cost", fmt = "d", color = colorCodes.LIFE, compPercent = true, lowerIsBetter = true, condFunc = function(v,o) return o.LifeHasCost end },
+		{ stat = "ManaCost", label = "Mana Cost", fmt = "d", color = colorCodes.MANA, pool = "Mana", compPercent = true, lowerIsBetter = true, condFunc = function(v,o) return o.ManaHasCost end, warnFunc = function(v,o) return (o.ManaCostWarning or (v > o.ManaUnreserved)) and "You do not have enough Mana to use a Selected Skill" end},
+		{ stat = "LifeCost", label = "Life Cost", fmt = "d", color = colorCodes.LIFE, pool = "Life", compPercent = true, lowerIsBetter = true, condFunc = function(v,o) return o.LifeHasCost end, warnFunc = function(v,o) return (o.LifeCostWarning or (v > o.LifeUnreserved)) and "You do not have enough Mana to use a Selected Skill" end },
```

Comment:
```suggestion
		{ stat = "LifeCost", label = "Life Cost", fmt = "d", color = colorCodes.LIFE, pool = "Life", compPercent = true, lowerIsBetter = true, condFunc = function(v,o) return o.LifeHasCost end, warnFunc = function(v,o) return (o.LifeCostWarning or (v > o.LifeUnreserved)) and "You do not have enough Life to use a Selected Skill" end },
```
Consider adding support for rage and energy shield cost as well. These warning might also be inaccurate with eldrich battery but I am not sure the best way to address this.

Reply:
Fixed by latest commit. Warnings for Energy shield and rage have been implemented.

Reply:
You forgot to commit the energy shield / rage implementation?

Reply:
Yup... Really need to pay more attention to what i'm doing.